x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java

```diff
@@ -127,15 +119,10 @@ private static CloseableHttpAsyncClient createAsyncClient(
     }
 
     public void start() {
-        if (status.compareAndSet(Status.CREATED, Status.STARTED)) {
-            client.start();
-        }
+        client.start();
```

Comment:
I looked at the apache implementation. It already has state tracking internally and we can call `start()` multiple times. Any additional `start()` calls will be ignored by the apache client.